astroid/nodes/scoped_nodes.py

```diff
@@ -1214,11 +1213,8 @@ def __init__(self, lineno=None, col_offset=None, parent=None):
         :type: dict(str, NodeNG)
         """
 
-        self.args = []
-        """The arguments that the function takes.
-
-        :type: Arguments or list
-        """
+        self.args: Arguments
```

Comment:
At some point we should probably enforce that `postinit` is actually called. Otherwise `self.args` might not be set when accessed. Task for another time. (It's somewhere on my long list.)

Reply:
This is already note the case.. Of the two places creating a `FunctioDef` only one does this. Oh well... 😄

Reply:
Which one do you mean. `raw_building`, `objectmodel`, and `rebuilder` are fine. Am I missing something? I didn't notice any errors while running the tests either.

Reply:
https://github.com/PyCQA/astroid/blob/9a7878a78a284571d3788afc42203089185cfcbf/astroid/raw_building.py#L127-L150

The `postinit` is called on `Arguments`, but the `FunctionDef.postinit` method is never called.

Reply:
At least `func.args` is still assigned. So it won't break (just yet).

Reply:
Something to fix in the future I guess. Seems like the `Lambda` and `FunctionDef` class could use a refactor anyway.

Reply:
Someday... Too many other things open at the moment 😪

Reply:
> Someday... Too many other things open at the moment 

Yeah let's finish up 2.11 first and the typing MR first. The code is getting better little by little and that's great, but let's merge what we did already. There are a lot of changes in those, and conflicts could become a problem.

Reply:
I expect to have #4980 ready for review in 5-10 minutes!